modules/heat_transfer/test/tests/transient_heat/transient_heat_derivatives.i

```diff
@@ -0,0 +1,98 @@
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 10
+    ny = 10
+  []
+[]
+
+[Variables]
+  [temp]
+    order = FIRST
+    family = LAGRANGE
+    initial_condition = 2
+  []
+[]
+
+[Kernels]
+  [heat]
+    type = HeatConduction
+    variable = temp
+  []
+
+  [ie]
+    type = HeatConductionTimeDerivative
+    variable = temp
+    specific_heat_dT = specific_heat_dT
+    density_name_dT = density_dT
+  []
+[]
+
+[Functions]
+  [spheat]
+    type = ParsedFunction
+    expression = 't^4'
+  []
+  [thcond]
+    type = ParsedFunction
+    expression = 'exp(t)'
+  []
+[]
+
+[BCs]
+  [bottom]
+    type = DirichletBC
+    variable = temp
+    boundary = 1
+    value = 4
+  []
+
+  [top]
+    type = DirichletBC
+    variable = temp
+    boundary = 2
+    value = 1
+  []
+[]
+
+[Materials]
+  [constant]
+    type = HeatConductionMaterial
+    thermal_conductivity_temperature_function = thcond
+    specific_heat_temperature_function = spheat
+    temp = temp
+  []
+  [density]
+    type = ParsedMaterial
+    property_name = density
+    coupled_variables = temp
+    expression = 'temp^3 + 2/temp'
+  []
+  [density_dT]
+    type = ParsedMaterial
+    property_name = density_dT
+    coupled_variables = temp
+    expression = '3 * temp^2 - 2/temp/temp'
+  []
```

Comment:
What a weird construct. Why not use a derivative parsed material here? The whole `xxx_dT` business predates the Derivative material stuff and really should be deprecated. We should not add new occurrences of this pattern.

Reply:
This is used just to make an overly complicated system to ensure that the jacobian has to be created correctly or it fails. See the other question for discussion on the dT method.